### PR TITLE
Mat 381 pull all

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

File: src/Application/Commands/PullMetaCampaignFromSF.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        parent::__construct();
+    }
+    public function configure(): void
+    {
+        $this->addArgument('metaCampaignSlug', InputArgument::REQUIRED);
+    }
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        // Steps:
+        // 1. Use SF API to get list of campaigns withi meta campaign by hitting e.g.
```
Comment (dorota-joanna) [inline src/Application/Commands/PullMetaCampaignFromSF.php:48]:
```suggestion
        // 1. Use SF API to get list of campaigns within meta campaign by hitting e.g.
```

**Review by dorota-joanna — APPROVED** [review]
LGTM

**Review by NoelLH — COMMENTED** [review]
Mostly looks good to me, minor things inline. Following up wider discussion about the API on Jira.

**Comment by bdsl** [comment]
@NoelLH I've updated this to fetch up to 10,000 campaigns from SF by paging through the API, instead of just taking the first page of 100. Still think we could make it quite a lot more efficient by changing the SF side to either return a list of IDs only or return multiple campaigns at once as needed by matchbot.

File: src/Domain/CampaignRepository.php [hunk 2]
```diff
@@ -98,6 +98,45 @@ public function findOneBySalesforceId(Salesforce18Id $salesforceId): ?Campaign
...
+        $count = count($campaignIds);
+
+        $i = 0;
+        foreach ($campaignIds as $id) {
+            $i++;
+            $this->logger->info("Fetching campaign $i of $count\n");
+            $existingCampaignInDB = $this->findOneBySalesforceId($id);
+
+            if ($existingCampaignInDB) {
+                $this->updateFromSf($existingCampaignInDB, withCache: false, autoSave: true);
```
Comment (NoelLH) [inline src/Domain/CampaignRepository.php:129]: Should we also add `$this->fundRepository->pullForCampaign($campaign);` around here as you suggested earlier? I think that would make the effect pretty much identical to what `UpdateCampaigns` does for already known ones.

Comment (bdsl) [reply src/Domain/CampaignRepository.php]: Yes I think so. I didn't want to make CampaignRepo depend on FundRepo so I've done it as a second phase after this loop completes in f80ca87efa0

Comment (NoelLH) [reply src/Domain/CampaignRepository.php]: Nice, looks good!

**Review by NoelLH — APPROVED** [review]

**Pull request merged**